include/oneapi/dpl/pstl/iterator_defs.h

```diff
@@ -50,30 +50,25 @@ struct __iterator_traits<_Tp*, void> : ::std::iterator_traits<_Tp*>
 {
 };
 
-// Make is_random_access_iterator not to fail with a 'hard' error when it's used in SFINAE with
-// a non-iterator type by providing a default value.
-template <typename _IteratorType, typename = void>
-struct __is_random_access_iterator_impl : ::std::false_type
-{
-};
+// Make is_random_access_iterator and is_forward_iterator not to fail with a 'hard' error when it's used in
+//SFINAE with a non-iterator type by providing a default value.
+template <typename _IteratorTag, typename... _IteratorTypes>
+auto
+__is_needed_iter(int)
```

Comment:
A comment about name - m.b. just `__is_iterator`? It is more readable. I think.

Reply:
the issue that I don't want to check if it's just an iterarator. I would like to check that it belongs to the particular category. Maybe it's better to rename it to `is_iterator_of`?

Reply:
yes, I'm agree with `is_iterator_of`